### PR TITLE
docs - pxf [cluster] sync now supports a delete option

### PR DESCRIPTION
**Pull request by lisakowen**
in this PR:   add the new option to the `pxf` and `pxf cluster` references pages (we don't currently have a dedicated topic for sync/reset)

questions:
1.  any reason an admin would want extra files in their standby/segment PXF user configs that aren't on the master?
2.  any upgrade implications - i.e. do we want to recommend that the user runs a sync with delete?

doc review site links:
- http://docs-lisa-pxf-syncdelete.cfapps.io/7-0/pxf/ref/pxf-cluster.html
- http://docs-lisa-pxf-syncdelete.cfapps.io/7-0/pxf/ref/pxf.html

### CODE REVIEW

File: gpdb-doc/markdown/pxf/ref/pxf-cluster.html.md.erb [hunk 1]
```diff
@@ -61,7 +61,15 @@ The `pxf cluster` utility command manages PXF on the master, standby master, and
...
 <dt>sync  </dt>
-<dd>Synchronize the PXF configuration from the master to the standby master and to all Greenplum Database segment hosts. If you have updated the PXF user configuration or added JAR files, you must also restart PXF after you synchronize the PXF configuration.</dd>
+<dd>Synchronize the PXF configuration (`$PXF_CONF`) from the master to the standby master and to all Greenplum Database segment hosts. By default, this command updates files on and copies files to the remote. You can instruct PXF to also delete files during the synchronization; see [Options](#options).</dd>
+<dd>If you have updated the PXF user configuration or added JAR files, you must also restart PXF after you synchronize the PXF configuration.</dd>
+
+## <a id="options"></a>Options
+
+The `pxf cluster sync` command takes the following option:
+
+<dt>-d | &#8211;&#8211;delete </dt>
```
Comment (oliverralbertini) [inline gpdb-doc/markdown/pxf/ref/pxf-cluster.html.md.erb]: Out of curiosity, why is the dash different for the two flags here?

File: gpdb-doc/markdown/pxf/ref/pxf.html.md.erb [hunk 2]
```diff
@@ -76,10 +76,13 @@ The `pxf init` command takes the following option:
 
 The `pxf reset` command takes the following option:
 
-<dt>-f | --force </dt>
+<dt>-f | &#8211;&#8211;force </dt>
```
Comment (oliverralbertini) [inline gpdb-doc/markdown/pxf/ref/pxf.html.md.erb]: Same question about the dashes here.

File: gpdb-doc/markdown/pxf/ref/pxf.html.md.erb [hunk 3]
```diff
@@ -76,10 +76,13 @@ The `pxf init` command takes the following option:
...
 <dd>Do not prompt before resetting the PXF service instance; reset without user interaction.</dd>
 
-The `pxf sync` command, run on the Greenplum Database master host, takes the following option:
+The `pxf sync` command, which you must run on the Greenplum Database master host, takes the following options:
+
+<dt>-d | &#8211;&#8211;delete </dt>
+<dd>Delete any files in the PXF user configuration on `<gphost>` that are not also present on the master host. If you specify this option, you must provide it on the command line before `<gphost>`.</dd>
```
Comment (oliverralbertini) [inline gpdb-doc/markdown/pxf/ref/pxf.html.md.erb]: Not the fault of this PR, but `gphost` is not an option, it's a mandatory argument for this command.

File: gpdb-doc/markdown/pxf/ref/pxf-cluster.html.md.erb (hunk 1 above)
Comment (lisakowen) [reply gpdb-doc/markdown/pxf/ref/pxf-cluster.html.md.erb]: markdown was rendering the dash-dash as a single, longer dash.  there may be a simpler way to render both dashes, this is using two unicode decimal representations so the dashes show up individually.

File: gpdb-doc/markdown/pxf/ref/pxf.html.md.erb (hunk 3 above)
Comment (lisakowen) [reply gpdb-doc/markdown/pxf/ref/pxf.html.md.erb]: the pxf references page markdown source has (consistently, i hope) been using the `<xx>` syntax to identify replaceable arguments, and the `[xx]` syntax for optional.

Comment (oliverralbertini) [reply gpdb-doc/markdown/pxf/ref/pxf.html.md.erb]: I was referring more to line 82, where it says:
>takes the following options:

File: gpdb-doc/markdown/pxf/ref/pxf-cluster.html.md.erb (hunk 1 above)
Comment (frankgh) [inline gpdb-doc/markdown/pxf/ref/pxf-cluster.html.md.erb]: I think the `-` renders better when using the HTML code &#8211;
```suggestion
<dt>&#8211;d | &#8211;&#8211;delete </dt>
```

File: gpdb-doc/markdown/pxf/ref/pxf.html.md.erb [hunk 4]
```diff
@@ -76,13 +76,16 @@ The `pxf init` command takes the following option:
 
 The `pxf reset` command takes the following option:
 
-<dt>-f | --force </dt>
+<dt>-f | &#8211;&#8211;force </dt>
```
Comment (frankgh) [inline gpdb-doc/markdown/pxf/ref/pxf.html.md.erb]: let's replace it here as well
```suggestion
<dt>&#8211;f | &#8211;&#8211;force </dt>
```

File: gpdb-doc/markdown/pxf/ref/pxf.html.md.erb [hunk 5]
```diff
@@ -76,13 +76,16 @@ The `pxf init` command takes the following option:
...
 <dd>Do not prompt before resetting the PXF service instance; reset without user interaction.</dd>
 
-The `pxf sync` command, run on the Greenplum Database master host, takes the following option:
+The `pxf sync` command, which you must run on the Greenplum Database master host, takes the following option and argument:
+
+<dt>-d | &#8211;&#8211;delete </dt>
```
Comment (frankgh) [inline gpdb-doc/markdown/pxf/ref/pxf.html.md.erb]:
```suggestion
<dt>&#8211;d | &#8211;&#8211;delete </dt>
```

File: gpdb-doc/markdown/pxf/ref/pxf-cluster.html.md.erb [hunk 6]
```diff
@@ -71,6 +79,12 @@ Stop the PXF service instance on all segment hosts:
 $ $GPHOME/pxf/bin/pxf cluster stop
 ```
 
+Synchronize the PXF configuration to the standby and all segment hosts, deleting files that do not exist on the master:
```
Comment (frankgh) [inline gpdb-doc/markdown/pxf/ref/pxf-cluster.html.md.erb]: master host maybe? 
```suggestion
Synchronize the PXF configuration to the standby and all segment hosts, deleting files that do not exist on the master host:
```

**Comment by lisakowen** [comment]
thanks for reviewing, @oliverralbertini and @frankgh .  what do you think about including a warning or note in the delete text that user would not want to use this option if their master and other pxf user configs are purposely different?  or is that obvious?

**Comment by frankgh** [comment]
@lisakowen I think the nodes need to be in sync with master. I don't think adding the warning will serve any purpose. I do not foresee a scenario where configurations are different on master and segment hosts, maybe @oliverralbertini or you can think of some situation where this might be needed?

**Comment by lisakowen** [comment]
@frankgh - the way we currently have the user set up their kerberos keytabs (see https://gpdb.docs.pivotal.io/6-4/pxf/pxf_kerbhdfs.html#proc_mit), they copy a host-specific keytab to `$PXF_CONF/keytabs/pxf.service.keytab` on each segment host.  i don't think we have them copy any keytabs to the master host.  in this scenario, would specifying `sync --delete` delete the keytabs on the segment hosts?

**Comment by oliverralbertini** [comment]
We specifically don't sync keytabs directory, just
${PXF_CONF}/{conf,libs,servers}

**Comment by lisakowen** [comment]
thanks for refreshing my memory about that, @oliverralbertini .  sounds like a warning is not needed.

**Comment by oliverralbertini** [comment]
Yeah, as long as there is no host-specific data in the new config stuff
that came along with the recent Kerberos work. I don't think there is,
though.

**Pull request merged**